File: app/src/main/java/com/example/notes/DataActions.java
package com.example.notes;

import java.util.List;
import java.util.UUID;

public interface DataActions {
    DataNote getData(int position);
    List<DataNote> getDataNoteList();
    int size();
    void addListElement(DataNote dataNote);
    void delete(int position);
    void clear();
    void updateElement(DataNote dataNote);
    void fetchList();
    void setUpdateListListener(UpdateDataListListener listener);
}

File: app/src/main/java/com/example/notes/DataActions.kt
package com.example.notes

interface DataActions {
    fun getData(position: Int): DataNote?
    fun getDataNoteList(): MutableList<DataNote?>?
    fun size(): Int
    fun addListElement(dataNote: DataNote?)
    fun delete(position: Int)
    fun clear()
    fun updateElement(dataNote: DataNote?)
    fun fetchList()
    fun setUpdateListListener(listener: UpdateDataListListener?)
}